heat/core/linalg/tests/test_solver.py

```diff
@@ -29,3 +29,104 @@ def test_cg(self):
             ht.linalg.cg(b, b, x0)
         with self.assertRaises(RuntimeError):
             ht.linalg.cg(A, b, A)
+
+    def test_lanczos(self):
+        # define positive definite matrix (n,n), split = 0
+        n = 100
+        A = ht.random.randn(n, n, dtype=ht.float64, split=0)
+        B = A @ A.T
+        # Lanczos decomposition with iterations m = n
+        V, T = ht.lanczos(B, m=n)
+        self.assertTrue(V.dtype is B.dtype)
+        self.assertTrue(T.dtype is B.dtype)
+        # V must be unitary
+        V_inv = ht.linalg.inv(V)
+        self.assertTrue(ht.allclose(V_inv, V.T))
+        # V T V.T must be = B, V transposed = V inverse
+        lanczos_B = V @ T @ V_inv
+        self.assertTrue(ht.allclose(lanczos_B, B))
+
+        # complex128
+        A = (
+            ht.random.rand(n, n, dtype=ht.float64, split=0)
+            + ht.random.rand(n, n, dtype=ht.float64, split=0) * 1j
+        )
+        A_conj = ht.conj(A)
+        B = A @ A_conj.T
+        # Lanczos decomposition with iterations m = n
+        V, T = ht.lanczos(B, m=n)
+        # V must be unitary
+        V_inv = ht.linalg.inv(V)
+        self.assertTrue(ht.allclose(V_inv, ht.conj(V).T))
+        # V T V* must be = B, V conjugate transpose = V inverse
+        lanczos_B = V @ T @ V_inv
+        self.assertTrue(ht.allclose(lanczos_B, B))
+
+        # single precision tolerance
+        if int(torch.__version__.split(".")[1]) == 13:
+            tolerance = 1e-3
+        else:
+            tolerance = 1e-4
```

Comment:
@Mystic-Slice re: #1028, here's a example of checking for the torch version, there might be more advanced ways that I don't know about though.